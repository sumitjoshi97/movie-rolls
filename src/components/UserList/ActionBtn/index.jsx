import React, { Component } from 'react'

export default class ActionBtn extends Component {
  render() {
    const style =
      this.props.active === this.props.name
        ? { backgroundColor: '#333', color: '#fff' }
        : null

    return (
      <button
        name='movie'
        onClick={() => this.props.onclick(this.props.name)}
        style={style}
      >
        {this.props.text} {this.props.name === 'movie' ? 'movies' : 'shows'}
      </button>
    )
  }
}
