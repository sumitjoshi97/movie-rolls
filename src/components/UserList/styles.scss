.user-list {
  display: flex;
  flex-direction: column;

  &__action-btns {
    position: absolute;
    top: 10rem;
    width: 100%;
    display: flex;
    justify-content: flex-start;
    padding: 0 4rem;

    & > * {
      padding: 2rem 3rem;
      background: #212121;
      color: #d8d8d8;
      border: none;
      font-size: 1.6rem;
      letter-spacing: 1px;
      text-transform: uppercase;
      margin-right: 4rem;
      cursor: pointer;

      &:focus,
      &:active {
        outline: none;
      }
    }
  }

  &__items {
    padding-top: 20rem;
  }
}
