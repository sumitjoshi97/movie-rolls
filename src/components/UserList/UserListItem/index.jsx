import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

import { setItemType } from '../../../actions'

import './styles.scss'

const UserListItem = ({ pageId, name, poster, type, ...props }) => {
  return (
    <div className='user-list-item'>
      <Link to={`/details/${pageId}`} onClick={() => props.setItemType(type)}>
        <img
          src={`https://image.tmdb.org/t/p/w185/${poster}`}
          alt={name}
          className='user-list-item__img'
        />
        <p className='user-list-item__name'>{name}</p>
      </Link>
    </div>
  )
}

export default connect(
  null,
  { setItemType }
)(UserListItem)
