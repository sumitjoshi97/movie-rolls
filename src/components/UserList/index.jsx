import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import List from '../List'

import './styles.scss'

export default class UserList extends PureComponent {
  static propTypes = {
    list: PropTypes.object
  }

  renderItems = () => {
    const { list } = this.props

    let movieList = []
    let tvList = []

    if (list) {
      for (let item in list) {
        if (list[item].type === 'movie') {
          movieList.push(list[item])
        } else {
          tvList.push(list[item])
        }
      }

      return (
        <>
          <List name='movies' type='movie' items={movieList} />
          <List name='TV shows' type='tv' items={tvList} />
        </>
      )
    }
  }

  render() {
    return <div className='user-list'>{this.renderItems()}</div>
  }
}
