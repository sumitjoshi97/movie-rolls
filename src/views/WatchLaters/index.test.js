import React from 'react'
import { shallow } from 'enzyme'

import { WatchLaters } from './index'
import UserList from '../../components/UserList'

describe('WatchLatersview unit tests', () => {
  describe('when watch-laters list is null', () => {
    const props = {
      watch: null
    }

    const watchLatersWrapper = shallow(<WatchLaters {...props} />)

    test('should render properly', () => {
      expect(watchLatersWrapper).toMatchSnapshot()
    })

    test('should render sorry-text', () => {
      expect(watchLatersWrapper.find('div.sorry-text')).toBeTruthy()
    })
  })

  describe('when WatchLatersis not null', () => {
    const props = {
      watch: {
        0: {
          id: '0'
        }
      }
    }
    const watchLatersWrapper = shallow(<WatchLaters {...props} />)
    test('should render UserList component', () => {
      expect(
        watchLatersWrapper.find({ list: props.watch }).find(UserList)
      ).toBeTruthy()
    })
  })
})
