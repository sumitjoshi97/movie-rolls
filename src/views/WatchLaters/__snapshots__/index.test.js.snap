// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchLatersview unit tests when watch-laters list is null should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WatchLaters
    watch={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="watch-later__header"
        >
          my watchlist
        </h2>,
        <h2
          className="sorry-text"
        >
          You don't have any yet, start adding now
        </h2>,
      ],
      "className": "watch-later",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "my watchlist",
          "className": "watch-later__header",
        },
        "ref": null,
        "rendered": "my watchlist",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You don't have any yet, start adding now",
          "className": "sorry-text",
        },
        "ref": null,
        "rendered": "You don't have any yet, start adding now",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="watch-later__header"
          >
            my watchlist
          </h2>,
          <h2
            className="sorry-text"
          >
            You don't have any yet, start adding now
          </h2>,
        ],
        "className": "watch-later",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "my watchlist",
            "className": "watch-later__header",
          },
          "ref": null,
          "rendered": "my watchlist",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You don't have any yet, start adding now",
            "className": "sorry-text",
          },
          "ref": null,
          "rendered": "You don't have any yet, start adding now",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
